index.js

```
@@ -116,82 +131,119 @@ app.use(Helmet())
-app.post('/v1/new', async function (req, res) {
-  const atomicAmount = toNumber(req.body.amount)
-  const callback = req.body.callback || false
-  const address = req.body.address || false
-  const name = req.body.name || false
-  const callerData = req.body.userDefined || {}
-  const confirmations = toNumber(req.body.confirmations)
-
-  if (!atomicAmount) {
-    logHTTPError(req, 'Invalid amount supplied')
-    return res.status(400).send()
-  }
+/* Validate the request from the user  */
+function validateRequest (
+  atomicAmount,
+  callback,
+  address,
+  confirmations,
+  callerData
+) {
+  return new Promise(function (resolve, reject) {
+    let walletRequest = {}
+
+    if (!atomicAmount) {
+      return reject('Invalid amount supplied')
+    }
 
-  const amount = (atomicAmount / Math.pow(10, Config.coinDecimals))
-  var cancelTimer
+    const amount = atomicAmount / Math.pow(10, Config.coinDecimals)
 
-  /* Validate that the caller has supplied a valid CryptoNote address
+    /* Validate that the caller has supplied a valid CryptoNote address
      for us to send funds to */
-  try {
-    cryptoUtils.decodeAddress(address)
-  } catch (e) {
-    logHTTPError(req, 'Invalid address supplied')
-    return res.status(400).send()
-  }
 
-  /* Verify that the caller supplied us with an acceptable callback
+    try {
+      cryptoUtils.decodeAddress(address)
+      walletRequest.address = address
+    } catch (e) {
+      return reject('Invalid address supplied')
+    }
+
+    /* Verify that the caller supplied us with an acceptable callback
      URL that we'll post back to later */
-  if (callback) {
-    if (callback.substring(0, 4).toLowerCase() !== 'http') {
-      logHTTPError(req, 'Invalid callback URL supplied')
-      return res.status(400).send()
+    if (callback) {
+      if (callback.substring(0, 4).toLowerCase() !== 'http') {
+        return reject('Invalid callback URL supplied')
```

reject should return an error. ie.

`reject(new Error('Invalid address supplied'))`